src/core/Slider/Slider.tsx

```diff
@@ -0,0 +1,494 @@
+   * 'odd-segments'- colored tracks shown in segments 1,3,5, etc.
+   *                 Default if number of thumbs values are even.
+   * 'even-segments'- colored tracks shown in segments 0,2,4, etc.
+   *                 Default if number of thumbs values are odd.
+   * @default 'auto'
+   */
+  trackDisplayMode?: TrackDisplayMode;
+  /**
+   * Step increment controls what values are allowed and the amount the value will
+   * change when left and right arrows are pressed when a Thumb has focus.
+   * @default 1
+   */
+  step?: number;
+  /**
+   * Forces control to be displayed in a disabled state where no interactive value
+   * changes are allowed.
+   *  @default false
+   */
+  disabled?: boolean;
+  /**
+   * Props to override default for showing a tooltip when Thumb is active or has focus.
+   */
+  tooltipProps?: Partial<Omit<TooltipProps, 'content' | 'children'>>;
+  /**
+   * Function that can generate tooltip content.
+   */
+  tooltipRenderer?: (val: number, step: number) => React.ReactNode;
+  /**
+   * Either an array of labels that will be placed under auto generated tick marks
+   * that are spaced evenly across width of Slider or a custom component that allows
+   * custom content to be placed in tick mark area below slider.
+   */
+  tickLabels?: React.ReactNode[] | JSX.Element;
+  /**
+   * Label for the minimum value. If undefined then the min
+   * value is shown. Use empty string for no label.
+   */
+  minLabel?: React.ReactNode;
+  /**
+   * Label for the maximum value. If undefined then the max
+   * value is shown. Use empty string for no label.
+   */
+  maxLabel?: React.ReactNode;
+  /**
+   * Additional props for container `<div>` that hold the slider rail, thumbs, and tracks.
+   */
+  railContainerProps?: React.HTMLAttributes<HTMLDivElement>;
+  /**
+   * Defines the allowed behavior when moving Thumbs when multiple Thumbs are
+   * shown. It controls if a Thumb movement should be limited to only move in
+   * the segments adjacent to the Thumb. Possible values:
+   * 'allow-crossing' - allows thumb to cross other thumbs. Default.
+   * 'inhibit-crossing'- keeps the thumb from crossing and separated by a step.
+   * @default 'inhibit-crossing'
+   */
+  thumbMode?: 'allow-crossing' | 'inhibit-crossing';
+  /**
+   * Callback that can provide additional props for `<div>` representing a thumb.
+   */
+  thumbProps?: (index: number) => React.HTMLAttributes<HTMLDivElement>;
+  /**
+   * Callback fired when the value(s) of the slider has changed. This will receive
+   * changes at the end of a slide as well as changes from clicks on rails and tracks.
+   */
+  onChange?: (values: ReadonlyArray<number>) => void;
+  /**
+   * Callback fired when the value(s) of the slider are internally updated during
+   * operations like dragging a Thumb. Use this callback with caution as a
+   * high-volume of updates will occur when dragging.
+   */
+  onUpdate?: (values: ReadonlyArray<number>) => void;
```

Comment:
I was reviewing tests and it made me wonder if this should be called `onChanged` or `onChangeCompleted` instead.

`onUpdate` is not self-explanatory.

Reply:
I have no objection to the name change, I used this name because react-compound-slider uses it and would eliminate me changing imodeljs-base slider code.